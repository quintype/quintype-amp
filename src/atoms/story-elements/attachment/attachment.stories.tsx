import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Attachment } from "./attachment";
import { config, textStory } from "../../../__fixtures__";
import { Layout } from "../../layout";

const sampleAttachmentElement = {
  description: "",
  "page-url": "/story/d66e8866-b815-4920-a79c-38e0b74c5b09/element/816fbbf8-e328-462e-aff3-484dd88d598d",
  type: "file",
  "family-id": "09572e29-b1f7-4026-a769-527a18c26cf0",
  title: "",
  id: "816fbbf8-e328-462e-aff3-484dd88d598d",
  "file-name":
    "Order of Honble CJ - Phased resumption of physical functioning of Courts at High Court of Gujarat -- Sd.pdf",
  url:
    "https://images.assettype.com/barandbench/2020-09/dfb40655-6bcb-428d-9869-6ffea81d9900/Order_of_Honble_CJ___Phased_resumption_of_physical_functioning_of_Courts_at_High_Court_of_Gujarat___.pdf",
  "s3-key":
    "barandbench/2020-09/dfb40655-6bcb-428d-9869-6ffea81d9900/Order_of_Honble_CJ___Phased_resumption_of_physical_functioning_of_Courts_at_High_Court_of_Gujarat___.pdf",
  "content-type": "application/pdf",
  metadata: {
    "file-size": 365267
  },
  subtype: "attachment"
};

storiesOf("Attachment", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <Attachment element={sampleAttachmentElement} />);
