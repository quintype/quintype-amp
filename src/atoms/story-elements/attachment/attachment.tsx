import React from "react";
import get from "lodash.get";
import { StoryElementProps } from "../types";
import styled from "styled-components";
import { media } from "../../../utils/media";
import { getLocalizedWord } from "../../../utils/localize-words/localization";
import { withStoryAndConfig } from "../../../context";

const AttachmentWrapper = styled.div`
  font-size: ${(props) => props.theme.font.size.m};
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  align-items: center;
  padding: 15px 25px;
  border: ${(props) => `1px solid ${props.theme.color.mono3}`};

  ${media.laptop`
    flex-direction: row;
  `}
`;
export const PreviewLink = styled.a`
  border: ${(props) => `1px solid ${props.theme.color.mono4}`};
  border-radius: 3px;
  padding: 10px 25px;
  color: unset;
  text-decoration: none;
  margin: 15px 0 0 0;

  ${media.laptop`
    margin: 0 0 0 15px;
  `}
`;

/**
 * Attachment is a story element and is displayed for type: file, subtype: attachment
 *
 * @param {Object} params object containing parameters
 * @param {Object} params.story story object
 * @param {Object} params.config config object
 * @param {Object} params.element the story element. This is same as the story element found in the story API response
 *
 *
 * @category StoryElements
 * @module FacebookElement
 * @component
 */

export const AttachmentBase = ({ element, story, config, counter }: StoryElementProps) => {
  const attachmentRender = get(config, ["opts", "render", "storyElementRender", "attachmentRender"], null);
  return attachmentRender ? (
    attachmentRender({ story, config, element, counter })
  ) : (
    <AttachmentWrapper>
      <div>{element["file-name"]}</div>
      <PreviewLink href={element.url} target="_blank" rel="noopener noreferrer" download="">
        {getLocalizedWord(config, "attachmentElementPreview", "Preview")}
      </PreviewLink>
    </AttachmentWrapper>
  );
};

export const Attachment = withStoryAndConfig(AttachmentBase);
