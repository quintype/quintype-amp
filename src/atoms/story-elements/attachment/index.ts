import { Attachment } from "./attachment";
export { Attachment };
