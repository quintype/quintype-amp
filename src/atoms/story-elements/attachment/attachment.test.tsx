import React from "react";
import { shallow } from "enzyme";
import { Attachment, AttachmentBase, PreviewLink } from "./attachment";

const sampleAttachmentElement = {
  description: "",
  "page-url": "/story/d66e8866-b815-4920-a79c-38e0b74c5b09/element/816fbbf8-e328-462e-aff3-484dd88d598d",
  type: "file",
  "family-id": "09572e29-b1f7-4026-a769-527a18c26cf0",
  title: "",
  id: "816fbbf8-e328-462e-aff3-484dd88d598d",
  "file-name":
    "Order of Honble CJ - Phased resumption of physical functioning of Courts at High Court of Gujarat -- Sd.pdf",
  url: "https://www.foo-bar/baz.pdf",
  "s3-key":
    "barandbench/2020-09/dfb40655-6bcb-428d-9869-6ffea81d9900/Order_of_Honble_CJ___Phased_resumption_of_physical_functioning_of_Courts_at_High_Court_of_Gujarat___.pdf",
  "content-type": "application/pdf",
  metadata: {
    "file-size": 365267
  },
  subtype: "attachment"
};

describe("Attachment", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Attachment element={sampleAttachmentElement} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should set correct attributes on anchor tag", () => {
    const wrapper = shallow(<AttachmentBase element={sampleAttachmentElement} />);
    expect(wrapper.find(PreviewLink).prop("href")).toBe("https://www.foo-bar/baz.pdf");
    expect(wrapper.find(PreviewLink).prop("target")).toBe("_blank");
    expect(wrapper.find(PreviewLink).prop("rel")).toBe("noopener noreferrer");
    expect(wrapper.find(PreviewLink).prop("download")).toBe("");
  });
});
