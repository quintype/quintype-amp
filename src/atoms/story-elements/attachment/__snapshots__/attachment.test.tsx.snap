// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachment should match snapshot 1`] = `
<styled.div>
  <div>
    Order of Honble CJ - Phased resumption of physical functioning of Courts at High Court of Gujarat -- Sd.pdf
  </div>
  <styled.a
    download=""
    href="https://www.foo-bar/baz.pdf"
    rel="noopener noreferrer"
    target="_blank"
  >
    Preview
  </styled.a>
</styled.div>
`;
